### ISSUE TITLE
FP: needless_raw_string_hashes on inner empty raw string

### ISSUE THREAD

**Issue by pms1969** [rust-lang/rust-clippy#11737]
### Summary

I have a situation where I'm including a raw string in a raw string... for testing, but I could also do this outside of a test.

Precisely, I have this:
```
        let input = r##"task test(image: rust:latest, execute: "") {r#""#}"##;
        check(input);
```

I have Pedantic lints enabled, and the empty raw string triggers the lint.  If I add a space to the inner raw string; or any other character; the lint does not fire. i.e.
```
        let input = r##"task test(image: rust:latest, execute: "") {r#"_"#}"##;
        check(input);
```

### Lint Name

needless_raw_string_hashes

### Reproducer

I tried this code:

```rust
        let input = r##"task test(image: rust:latest, execute: "") {r#""#}"##;
        check(input);
```

I saw this happen:

```
   |
73 |         let input = r##"task test(image: rust, execute: "") {r#""#}"##;
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `r#"task test(image: rust, execute: "") {r#""#}"#`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_raw_string_hashes
   = note: `-D clippy::needless-raw-string-hashes` implied by `-D warnings`

```

I expected to see this happen:
I do not expect the lint to trigger in this situation.

### Version

```text
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: aarch64-apple-darwin
release: 1.73.0
LLVM version: 17.0.2
```

**Comment by pms1969** [comment]
I think the error is [here](https://github.com/rust-lang/rust-clippy/blob/7d34406015fce08031d0986dec0a4689a3cc407e/clippy_lints/src/raw_strings.rs#L117) and that this should just remove the conditional
```
b'"' => (following_quote, req) = (true, 1),
```

I have no idea what the impact of this would be on the rest of the lint tho so of course you should treat that with a sampling of salt.
